src/fsharp/ReferenceLoading.PaketHandler.fs

```diff
@@ -0,0 +1,232 @@
+                   (File.ReadAllLines(workingDirSpecFile.FullName) |> Array.toList) = packageManagerTextLines && 
+                   File.Exists loadScript
+                then 
+                    printfn "skipping running package resolution... already done that" 
+                    Solved(loadScript,additionalIncludeFolders())
+                else
+                    try File.Delete(loadScript) with _ -> ()
+                    let toolPath = alterToolPath toolPath
+                    File.WriteAllLines(workingDirSpecFile.FullName, packageManagerTextLines)
+                    printfn "running package resolution in '%s'..." workingDir
+                    let startInfo = 
+                        System.Diagnostics.ProcessStartInfo(
+                            FileName = toolPath,
+                            WorkingDirectory = workingDir, 
+                            RedirectStandardOutput = true,
+                            RedirectStandardError = true,
+                            Arguments = getCommand targetFramework rootDir,
+                            CreateNoWindow = true,
+                            UseShellExecute = false)
+                
+                    use p = new System.Diagnostics.Process()
+                    let errors = System.Collections.Generic.List<_>()
+                    let log = System.Collections.Generic.List<_>()
+                    p.StartInfo <- startInfo
+                    p.ErrorDataReceived.Add(fun d -> if d.Data <> null then errors.Add d.Data)
+                    p.OutputDataReceived.Add(fun d -> if d.Data <> null then log.Add d.Data)
+                    p.Start() |> ignore
+                    p.BeginErrorReadLine()
+                    p.BeginOutputReadLine()
+                    p.WaitForExit()
+
+                    printfn "done running package resolution..."
+                    if p.ExitCode <> 0 then
+                        let msg = String.Join(Environment.NewLine, errors)
+                        PackageResolutionFailed(toolPath, workingDir, msg)
+                    else
+                        printfn "package resolution completed at %A" System.DateTimeOffset.UtcNow
+                        Solved(loadScript,additionalIncludeFolders())
+
+    let getLoadScript baseDir packageManagerLoadScriptSubDirectory loadScriptName =
+      System.IO.Path.Combine(baseDir, packageManagerLoadScriptSubDirectory, loadScriptName)
+
+    /// Resolves absolute load script location: something like
+    /// baseDir/.paket/load/scriptName
+    /// or
+    /// baseDir/.paket/load/frameworkDir/scriptName 
+    let GetPaketLoadScriptLocation baseDir optionalFrameworkDir scriptName =
+      let paketLoadFolder = System.IO.Path.Combine(Internals.PM_DIR,"load")
+      getLoadScript
+        baseDir
+        (match optionalFrameworkDir with | None -> paketLoadFolder | Some frameworkDir -> System.IO.Path.Combine(paketLoadFolder, frameworkDir))
+        scriptName
+
+
+module GenericHandler =
```

Comment:
@dsyme "putting more code in CompileOps/fsi considered harmful" :)

For someone trying to contribute, dealing with those huge compact files is really daunting, if we use small files (and some folder structure) I think the codebase would tend to become more manageable.

In my day job, I also deal with large files at times, when I see a change just adding some line of code, which could instead have been put in a separate component with only small insertion point fitting better the level at integration point, I feel the change was done in haste without considering that option.

I'm not advocating the C#/Java approach here though, maybe we can consider a PragmaHandling.fs module with the code you are pointing.